### PR TITLE
[BugFix] Fix multi-processing bug in data simulator

### PR DESCRIPTION
**Pull request by tango4j**
# What does this PR do ?

Fix multi-processing bug in data simulator

**Collection**: 
ASR 

# Changelog 
- Add specific line by line info of high level changes in this PR.
- Changed silence sampling logic to follow overlap sampling logic
- Changed perturbator config structure to include `add_xxx_aug` in side `xxxxxx_augmentor`
- Some comment changes
- Fixed annotation file writing mechanism to fix multi-processing issue.
- Added more unit-tests for util functions 

# Usage

```python
This script creates a synthetic diarization session using the provided audio dataset with ctm files.
Usage:
  python <NEMO_ROOT>/tools/speech_data_simulator/multispeaker_simulator.py  \
    num_workers=10 \
    data_simulator.random_seed=42 \
    data_simulator.manifest_filepath=manifest_with_alignment_file.json \
    data_simulator.outputs.output_dir=./simulated_data \
    data_simulator.outputs.output_filename=sim_spk2_sess20 \
    data_simulator.session_config.num_sessions=1000 \
    data_simulator.session_config.num_speakers=2 \
    data_simulator.session_config.session_length=20 \
    data_simulator.background_noise.add_bg=False \
    data_simulator.background_noise.background_manifest=background_noise.json \
    data_simulator.background_noise.snr=40 \

Check out parameters in ./conf/data_simulator.yaml.
```

# Before your PR is "Ready for review"
**Pre checks**:
- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [x] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [x] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Review by stevehuang52 — DISMISSED** [review]
I've tested the code and found the silence/overlap distributions satisfactory enough, thanks for the great work~!

**Comment by tango4j** [comment]
Please do not merge this until we test tutorial notebooks. Thanks.

File: nemo/collections/asr/data/data_simulation.py [hunk 1]
```diff
@@ -162,19 +165,21 @@ class MultiSpeakerSimulator(object):
...
-      white_noise:
-        prob (float): 1.0  # probability of adding white noise
-        min_level: -90
-        max_level: -46
+      add_sess_aug: (bool) set True to enable audio augmentation on the whole session (Default: False)
+      segmentor:
+        white_noise:
+            prob (float): Probability of adding white noise (Default: 1.0)
+            min_level (float): minimum gain in terms of dB
+            max_level (float): minimum gain in terms of dB
```
[the remark this answers is not in the document]
Comment (tango4j) [reply nemo/collections/asr/data/data_simulation.py]: fixed

**Review by stevehuang52 — APPROVED** [review]
LGTM

**Pull request merged**